### PR TITLE
MINIFICPP-1567 enable linter checks in extensions (part 4)

### PR DESCRIPTION
**Pull request by martinzink**
enable rocksdb-repos linter
enable jni linter
enable script linter
enable sftp linter
enable sensor linter

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by lordgamez — APPROVED** [review]

File: extensions/rocksdb-repos/DatabaseContentRepository.h [hunk 1]
```diff
@@ -38,7 +40,7 @@ namespace repository {
 class StringAppender : public rocksdb::AssociativeMergeOperator {
  public:
   // Constructor: specify delimiter
-  explicit StringAppender() = default;
+  StringAppender() = default;
```
Comment (fgerlits) [inline extensions/rocksdb-repos/DatabaseContentRepository.h:43]: minor, but this comment looks stale; I would remove both lines, as the compiler will generate a default constructor anyway

File: extensions/rocksdb-repos/ProvenanceRepository.h [hunk 2]
```diff
@@ -156,7 +162,7 @@ class ProvenanceRepository : public core::Repository, public std::enable_shared_
       std::string key = it->key().ToString();
       if (store.size() >= max_size)
         break;
-      if (eventRead->DeSerialize((uint8_t *) it->value().data(), (int) it->value().size())) {
+      if (eventRead->DeSerialize(reinterpret_cast<const uint8_t *>(it->value().data()), static_cast<int>(it->value().size()))) {
```
Comment (fgerlits) [inline extensions/rocksdb-repos/ProvenanceRepository.h:165]: this cast to `int` looks dangerous (we may be silently dropping the high-order bits), and it is also unnecessary, as the second parameter of `DeSerialize()` is of type `size_t` [also in a few places further down in this file]
```suggestion
      if (eventRead->DeSerialize(reinterpret_cast<const uint8_t *>(it->value().data()), it->value().size())) {
```

File: extensions/script/python/PyProcCreator.h [hunk 3]
```diff
@@ -56,6 +58,7 @@ class PythonObjectFactory : public core::DefautObjectFactory<minifi::python::pro
   /**
    * Create a shared pointer to a new processor.
    */
+
```
Comment (fgerlits) [inline extensions/script/python/PyProcCreator.h:61]: very minor, but why did you add this empty line?

I would remove it, and I would also remove the comment, and all four doxygen comments from the `create()` functions, as they are 50% incorrect and 100% unnecessary

File: extensions/sensors/SensorBase.h [hunk 4]
```diff
@@ -64,17 +61,17 @@ class SensorBase : public core::Processor {
...
+    explicit WriteCallback(std::string data)
+        : data_{std::move(data)} {
+    }
+    std::string data_;
+    int64_t process(const std::shared_ptr<io::BaseStream>& stream) override {
+      if (data_.empty()) return 0;
+      const auto write_ret = stream->write(reinterpret_cast<const uint8_t*>(data_.data()), data_.size());
+      return io::isError(write_ret) ? -1 : gsl::narrow<int64_t>(write_ret);
+    }
+};
```
Comment (fgerlits) [inline extensions/sensors/SensorBase.h:74]: this should be indented; I would also add a blank line after it
```suggestion
  };

```

File: extensions/sftp/processors/ListSFTP.cpp [hunk 5]
```diff
@@ -59,73 +59,85 @@ namespace nifi {
...
+    ->withAllowableValues<std::string>({LISTING_STRATEGY_TRACKING_TIMESTAMPS, LISTING_STRATEGY_TRACKING_ENTITIES})
+    ->withDefaultValue(LISTING_STRATEGY_TRACKING_TIMESTAMPS)->build());
+
+core::Property ListSFTP::RemotePath(core::PropertyBuilder::createProperty("Remote Path")
+    ->withDescription("The fully qualified filename on the remote system")
+    ->isRequired(false)->supportsExpressionLanguage(true)->build());
+
+core::Property ListSFTP::SearchRecursively(core::PropertyBuilder::createProperty("Search Recursively")
+    ->withDescription("If true, will pull files from arbitrarily nested subdirectories; "
                                                                                  "otherwise, will not traverse subdirectories")
```
Comment (fgerlits) [inline extensions/sftp/processors/ListSFTP.cpp:74]: this should be indented less:
```suggestion
    ->withDescription("If true, will pull files from arbitrarily nested subdirectories; "
                      "otherwise, will not traverse subdirectories")
```

File: extensions/sftp/processors/SFTPProcessorBase.cpp [hunk 6]
```diff
@@ -48,59 +48,74 @@ namespace nifi {
...
+    ->withDescription("Indicates whether or not to send a single Keep Alive message when SSH socket times out")
+    ->isRequired(true)->withDefaultValue<bool>(true)->build());
+
+core::Property SFTPProcessorBase::ProxyType(core::PropertyBuilder::createProperty("Proxy Type")
+    ->withDescription("Specifies the Proxy Configuration Controller Service to proxy network requests. If set, it supersedes proxy settings configured per component. "
+                       "Supported proxies: HTTP + AuthN, SOCKS + AuthN")
+    ->isRequired(false)
+    ->withAllowableValues<std::string>({PROXY_TYPE_DIRECT,
                                             PROXY_TYPE_HTTP,
                                             PROXY_TYPE_SOCKS})
```
Comment (fgerlits) [inline extensions/sftp/processors/SFTPProcessorBase.cpp:101]: these should line up:
```suggestion
    ->withAllowableValues<std::string>({PROXY_TYPE_DIRECT,
                                        PROXY_TYPE_HTTP,
                                        PROXY_TYPE_SOCKS})
```

File: extensions/sftp/tests/FetchSFTPTests.cpp [hunk 7]
```diff
@@ -142,7 +142,7 @@ class FetchSFTPTestsFixture {
     std::fstream file;
     std::stringstream ss;
     ss << src_dir << "/vfs/" << relative_path;
-    utils::file::FileUtils::create_dir(utils::file::FileUtils::get_parent_path(ss.str()));  // TODO
+    utils::file::FileUtils::create_dir(utils::file::FileUtils::get_parent_path(ss.str()));  // TODO(_)
```
Comment (fgerlits) [inline extensions/sftp/tests/FetchSFTPTests.cpp:145]: I doubt anyone knows what this TODO refers to, so I think we should remove it rather than make it cpplint-compliant

also, `utils::file::FileUtils::` is the same as `utils::file::`, so we can remove `FileUtils`; maybe that is what the TODO was about?

```suggestion
    utils::file::create_dir(utils::file::get_parent_path(ss.str()));
```

**Review by fgerlits — APPROVED** [review]
